renku/cli/_providers/dataverse.py

```diff
@@ -0,0 +1,277 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright 2019 - Swiss Data Science Center (SDSC)
+# A partnership between École Polytechnique Fédérale de Lausanne (EPFL) and
+# Eidgenössische Technische Hochschule Zürich (ETHZ).
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Dataverse API integration."""
+import pathlib
+import re
+import urllib
+import urllib.parse as urlparse
+
+import attr
+import requests
+
+from renku.cli._providers.api import ExporterApi, ProviderApi
+from renku.cli._providers.doi import DOIProvider
+from renku.models.datasets import Dataset, DatasetFile
+from renku.utils.doi import extract_doi, is_doi
+
+DATAVERSE_API_PATH = 'api'
+
+DATAVERSE_VERSION_API = 'info/version'
+DATAVERSE_METADATA_API = 'datasets/export'
+DATAVERSE_FILE_API = 'access/datafile/:persistentId/'
+DATAVERSE_EXPORTER = 'schema.org'
+
+
+def check_dataverse_uri(url):
+    """Check if an URL points to a dataverse instance."""
+    url_parts = list(urlparse.urlparse(url))
```

Comment:
sorry, I didn't notice this before - why turn the parsed URL into a list? It should be much cleaner to just refer to the parts of the URL by name.

Reply:
`ParseResult` properties can't be set directl:
```
>>> from urllib.parse import urlparse
>>> r = urlparse("http://www.google.com/search?q=wer")
>>> r
ParseResult(scheme='http', netloc='www.google.com', path='/search', params='', query='q=wer', fragment='')
>>> r.path = "bla"
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: can't set attribute
```

It can be side by using the private method `_replace` as discussed here: https://stackoverflow.com/questions/21628852/changing-hostname-in-a-url though I'm not sure if using a private method is any cleaner. It is a bit of a weird design decision by the urllib developers.

Reply:
Ah I didn't realize - that's pretty weird indeed. Also not possible to turn it into a dictionary...